src/main/scala/beam/agentsim/events/ParkingUtilityEvent.scala

```diff
@@ -0,0 +1,124 @@
+package beam.agentsim.events
+
+import java.util
+
+import beam.agentsim.agents.vehicles.{BeamVehicle, BeamVehicleType}
+import beam.agentsim.infrastructure.charging.ChargingPointType
+import beam.agentsim.infrastructure.parking.{ParkingMNL, ParkingType}
+import beam.agentsim.infrastructure.parking.ParkingZoneSearch.ParkingZoneSearchStats
+import org.matsim.api.core.v01.events.Event
+
+case class ParkingUtilityEvent(
+  driverId: String,
+  beamVehicle: Option[BeamVehicle],
+  activityType: String,
+  parkingDuration: Double,
+  agentVoT: Double,
+  parkingZoneSearchStats: ParkingZoneSearchStats,
+  selectedStallPrice: Double,
+  selectedStallParkingType: ParkingType,
+  selectedStallChargingPointType: Option[ChargingPointType],
+  tick: Double = -1 // todo maybe we need the currentTick? If yes we have to get it into ZonalParkingManager
+) extends Event(tick)
+    with ScalaEvent {
+
+  import ParkingUtilityEvent._
+
+  private lazy val sampledStallsChargingTypeDist: String = "[" + parkingZoneSearchStats.sampledStallsChargingTypes
```

Comment:
regarding `sampledStallsChargingTypeDist | sampledStallsParkingTypeDist | sampledStallsCostsDist`,

i'm a little worried about this approach and how much clutter it will add, and how it won't be easy to convert for use in data analysis. i know, when i started logging parking utility data, i was grabbing this kind of stuff, but it was mostly for debug reasons. maybe we leave these out from the Event?

i think if we want to study this kind of data, we should make additional events for it. a `SampledStallsChargingTypeDist` event, etc., though i would want to know what kinds of analysis tasks that would support, i.e., what questions we are asking to necessitate grabbing that data.

Reply:
and, i also see that this would mean removing like 50-60 lines of your work, so, let me say that i appreciate your efforts on this. i just don't know that we need that data or would want it in that form?

Reply:
Agree that it won't be easy to convert this kinda data into a usable format for data analysis. I left it inside mainly because of the reason that I assumed we initially wanted this event for debugging purposes. 

I think a `SampledStallsChargingTypeDist`event wouldn't add much value for data analysis - only for debugging purposes and to understand what's going on in the sampling. So maybe we just drop it for now?

Reply:
Okay I just dropped it for now in it's own commit 

`ParkingZoneSearch`still includes `parkingZoneIdsSampled` which hasn't been mentioned in your review as dropping candidate. As this is part of the existing code, let me know if you think this param adds some value on data analysis steps and I'll include it into the event or drop it otherwise as well.